node/pkg/logscribe/app.go

```diff
@@ -87,3 +148,125 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
+		logMap[log.Service][hash] = append(logMap[log.Service][hash], log)
+	}
+
+	for service, hashLogPairs := range logMap {
+		pairs := make([]HashLogPairs, 0, len(hashLogPairs))
+		for hash, logs := range hashLogPairs {
+			pairs = append(pairs, HashLogPairs{hash, logs})
+		}
+		sort.Slice(pairs, func(i, j int) bool {
+			return len(pairs[i].logs) > len(pairs[j].logs)
+		})
+
+		topOccurrences := pairs
+		if len(pairs) > topOccurrencesForGithubIssue {
+			topOccurrences = pairs[:topOccurrencesForGithubIssue]
+		}
+		processedLogs[service] = make([]LogInsertModelWithIDWithCount, 0, len(topOccurrences))
+		for _, pair := range topOccurrences {
+			// all logs in pair.logs are same (due to hashing), so only need to store one
+			processedLogs[service] = append(
+				processedLogs[service],
+				LogInsertModelWithIDWithCount{OccurrenceCount: len(pair.logs), LogInsertModelWithID: pair.logs[0]},
+			)
+		}
+	}
+
+	return processedLogs
+}
+
+func fetchDeleteLogs(ctx context.Context) ([]LogInsertModelWithID, error) {
+	logs, err := db.QueryRows[LogInsertModelWithID](ctx, readDeleteLogsQuery, nil)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to read logs")
+		return nil, err
+	}
+	return logs, nil
+}
+
+func hashLog(log LogInsertModelWithID) string {
+	hash := sha256.New()
+	hash.Write([]byte(log.Service))
+	hash.Write([]byte(fmt.Sprintf("%d", log.Level)))
+	hash.Write([]byte(log.Message))
+	hash.Write(log.Fields)
+	return hex.EncodeToString(hash.Sum(nil))
+}
+
+func (a *App) createGithubIssue(ctx context.Context, processedLogs map[string][]LogInsertModelWithIDWithCount) {
+	if a.githubOwner == "test" {
+		return
+	}
+
+	issueCount := 0
+	for service, logs := range processedLogs {
+		var logsJson string
+		for _, entry := range logs {
+			hash := hashLog(entry.LogInsertModelWithID)
+			res, err := db.QueryRow[Count](ctx, logAlreadyProcessedQuery, map[string]any{
+				"hash": hash,
+			})
+			if err != nil {
+				log.Error().Err(err).Msg("Failed to check if log already processed")
+				continue
+			}
+			if res.Count > 0 {
+				log.Debug().Msg("Log already processed, skipping creation of github issue")
+				continue
+			}
+
+			entryJson, err := json.MarshalIndent(entry, "", "  ")
+			if err != nil {
+				log.Error().Err(err).Msg("Failed to marshal log")
+				return
+			}
+			logsJson += string(entryJson) + "\n"
+			formattedBody := "```go\n" + logsJson + "\n```"
+			issueRequest := &github.IssueRequest{
+				Title:  github.String(fmt.Sprintf("[%s] %s", service, entry.Message)),
+				Body:   github.String(formattedBody),
+				Labels: &[]string{"bug"},
+			}
+
+			_, resp, err := a.githubClient.Issues.Create(ctx, a.githubOwner, a.githubRepo, issueRequest)
+			if err != nil || resp.StatusCode != http.StatusCreated {
+				log.Error().Err(err).Msg("Failed to create github issue")
+			}
+
+			err = db.QueryWithoutResult(ctx, "INSERT INTO processed_logs (log_hash) VALUES (@hash)", map[string]any{
```

Comment:
let's also put the query into const variable
and it'd be nice if it can also be done in bulk